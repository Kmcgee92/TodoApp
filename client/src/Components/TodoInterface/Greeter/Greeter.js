import React from "react";
//mui components
import Toolbar from "@material-ui/core/Toolbar";

const Greeter = ({ activeUser, modalOpen, setModalOpen, classes }) => {
  return (
    <>
      <Toolbar />
      {!Object.keys(activeUser).length && !modalOpen ? (
        <div data-test="greeting">
          <span className={classes.signupContent}>
            need to create an account?&nbsp;&nbsp;
            <span
              data-test="signupModalOpen"
              className={classes.signupToggleParent}
              onClick={() => setModalOpen(true)}
            >
              <span className={classes.signupToggle}>Signup</span>
            </span>
          </span>
        </div>
      ) : null}
    </>
  );
};

export default Greeter;
