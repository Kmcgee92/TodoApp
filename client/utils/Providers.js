import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
// provide redux and apollo
import { Provider } from "react-redux";
import { MockedProvider } from "@apollo/client/testing";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

export default function ComponentWithProviders(
  ComponentToBeTested,
  props,
  initialReduxMocks,
  initialApolloMocks
) {
  let mocks = initialApolloMocks || [];
  let store = mockStore(
    
    initialReduxMocks || { auth: { activeUser: {}, error: false } }
  
  );
  return (
    <MockedProvider mocks={mocks}>
      <Provider store={store}>
        <ComponentToBeTested {...props} />
      </Provider>
    </MockedProvider>
  );
}
