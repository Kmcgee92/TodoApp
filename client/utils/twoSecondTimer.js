export default function twoSecondTmer() {
  return new Promise((resolve) => {
    setTimeout(resolve, 2000);
  });
}
