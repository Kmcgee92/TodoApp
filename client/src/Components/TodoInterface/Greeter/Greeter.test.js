import React from "react";
import { mount, shallow } from "enzyme";
import toJSON from "enzyme-to-json";
// component to test
import Greeter from "./Greeter";
import { isDescribable } from "graphql-tools";
// utils
import { findByTestAtrr } from "../../../../utils/index";
import { ExpansionPanelActions, jssPreset } from "@material-ui/core";

describe("<Greeter />", () => {
  let greeter;

  let mockProps = {
    activeUser: {},
    modalOpen: false,
    setModalOpen: jest.fn(),
    classes: {},
  };
  beforeEach(() => {
    greeter = shallow(<Greeter {...mockProps} />);
  });
  it("Should Render", () => {
    expect(greeter).toHaveLength(1);
    expect(toJSON(greeter)).toMatchSnapshot();
  });
  it("Should Call Props Function 'SetModalOpen' utilizing onClick", () => {
    findByTestAtrr(greeter, "signupModalOpen").simulate("click");
    expect(mockProps.setModalOpen).toBeCalledTimes(1);
  });
  it("Should Not Render data-test='greeting' When There is an Active User", () => {
    mockProps.activeUser = { mockUser: {} };
    greeter = shallow(<Greeter {...mockProps} />);
    expect(findByTestAtrr(greeter, "greeting")).toHaveLength(0);
  });
});
