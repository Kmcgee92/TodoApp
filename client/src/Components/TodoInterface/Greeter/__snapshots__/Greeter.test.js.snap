// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Greeter /> Should Render 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Toolbar)) />
  <div
    data-test="greeting"
  >
    <span>
      need to create an account?  
      <span
        data-test="signupModalOpen"
        onClick={[Function]}
      >
        <span>
          Signup
        </span>
      </span>
    </span>
  </div>
</Fragment>
`;
